python/pyarrow/scalar.pxi

```diff
@@ -902,10 +902,7 @@ cdef class ExtensionScalar(Scalar):
         """
         Return this scalar as a Python object.
         """
-        if self.value is None:
-            return None
-        else:
-            return self.type.scalar_as_py(self.value)
+        return self.value if self.value is not None else None
```

Comment:
```suggestion
        return self.value.as_py() if self.value is not None else None
```

? (it is a bit suspicious that the tests are not failing because of this)

Reply:
Changed it around a bit, but it's the same logic really.